### PR TITLE
Add videolink attribute to recipes

### PR DESCRIPTION
**Pull request by TomaszLaski**

### CODE REVIEW

File: src/components/RecipeForm/RecipeForm.tsx [hunk 1]
```diff
@@ -1,41 +1,25 @@
...
+  POSITIVE_NUMBER_REGEXP,
+  PRICE_REGEXP,
+  URL_REGEXP,
+} from 'helpers/validation';
 
-export const RecipeForm: React.FC = () => {
-  const { control, handleSubmit, getValues } = useForm<RecipeModel>({
+type FormProps = {
+  onSubmit: (data: RecipeModel) => Promise<void>;
+  defaultValue?: any;
```
Comment (Aliath) [inline src/components/RecipeForm/RecipeForm.tsx:15]: Get rid of `any`

File: src/components/AddRecipe/AddRecipe.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      <RecipeForm onSubmit={onSubmit} defaultValue="" />
+      <Snackbar
+        open
+        onClose={handleClose}
+        anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
+      >
+        <div>
+          {responseState === 'success' && (
+            <Alert onClose={handleClose} severity="success">
+              The recipe was created!.
```
Comment (Aliath) [inline src/components/AddRecipe/AddRecipe.tsx:43]: Remove exclamation mark

File: src/components/RecipeUpdate/RecipeUpdate.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+export const RecipeUpdate: React.FC<UpdateProps> = ({
+  setRecipeUpdated,
+  dataRecipes,
+  setDataRecipes,
+}) => {
+  const classes = useStyles();
+  const [modalOpen, setModalOpen] = React.useState(false);
+  const { id } = useParams<{ id: string }>();
+  const [responseState, setResponseState] = useState<
```
Comment (Aliath) [inline src/components/RecipeUpdate/RecipeUpdate.tsx:59]: Change namings to `updateResponseState`, `fetchResponseState`

File: src/components/RecipeUpdate/RecipeUpdate.tsx [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+      try {
+        const recipes = await getRecipe(id);
+        setDataRecipes(recipes);
+        setFetchState('success');
+      } catch {
+        setFetchState('failure');
+      }
+    }
+    fetchData();
+  }, [id]);
```
Comment (Aliath) [inline src/components/RecipeUpdate/RecipeUpdate.tsx:97]: `React Hook useEffect has a missing dependency: 'setDataRecipes'. Either include it or remove the dependency array. If 'setDataRecipes' changes too often, find the parent component that defines it and wrap that definition in useCallback`

File: src/components/RecipeUpdate/RecipeUpdate.tsx [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+        <Snackbar
+          open={responseState === 'success' || responseState === 'failure'}
+          anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
+          autoHideDuration={1500}
+          onClose={handleClose}
+        >
+          <div>
+            {responseState === 'success' && (
+              <Alert onClose={handleClose} severity="success">
+                The recipe was updated!.
```
Comment (Aliath) [inline src/components/RecipeUpdate/RecipeUpdate.tsx:125]: Remove exclamation mark

File: src/components/RecipeDetails/RecipeDetails.tsx [hunk 6]
```diff
@@ -132,6 +141,11 @@ export const RecipeDetails: React.FC = () => {
         >
           Go back to list
         </Button>
+        <RecipeUpdate
+          setRecipeUpdated={setRecipeUpdated}
+          dataRecipes={dataRecipes}
+          setDataRecipes={setDataRecipes}
```
Comment (Aliath) [inline src/components/RecipeDetails/RecipeDetails.tsx:147]: Change it to `onUpdate={onUpdate}` and then fetch details once again

Comment (Aliath) [reply src/components/RecipeDetails/RecipeDetails.tsx:147]: Btw. `setDataRecipes` -> `setRecipeDetails`

File: src/components/RecipeDetails/RecipeDetails.tsx [hunk 7]
```diff
@@ -69,6 +75,9 @@ export const RecipeDetails: React.FC = () => {
     'loading' | 'success' | 'error'
   >('loading');
   const { id } = useParams<{ id: string }>();
+  const [recipeUpdated, setRecipeUpdated] = useState<'updated' | 'waiting'>(
```
Comment (Aliath) [inline src/components/RecipeDetails/RecipeDetails.tsx:78]: Keep it similar across all request states - `recipeUpdateState<'loading' | 'idle'>`

**Review by Aliath — CHANGES_REQUESTED** [review]

File: src/components/RecipeForm/RecipeForm.tsx (hunk 1 above)
Comment (TomaszLaski) [reply src/components/RecipeForm/RecipeForm.tsx:15]: Unfortunately, I lack TS skills to fix this problem despite several hours of trying

**Review by Aliath — APPROVED** [review]

**Pull request merged**